### PR TITLE
Improved patching mechanism and printing logs

### PR DESCRIPTION
**Pull request by lwesiers**

### CODE REVIEW

File: cmake/modules/CMakeFunctions.cmake [hunk 1]
```diff
@@ -39,6 +39,51 @@ macro(use_eh val)
     endif()
 endmacro(use_eh)
 
+# Reads hash-commit from backported patch
+# This function assumes that each of files starts with (for example):
+# From 1a400928bf8fc86fa0f062524c25d0985c94ac6f Mon Sep 17 00:00:00 2001
+function(get_backport_patch_hash patch_path patch_hash)
```
Comment (hewj03) [inline cmake/modules/CMakeFunctions.cmake:45]: can we use "git rev-parse " to get commit id?

File: cmake/modules/CMakeFunctions.cmake [hunk 2]
```diff
@@ -39,6 +39,51 @@ macro(use_eh val)
...
+        set(patch_in_branch False PARENT_SCOPE) # The patch is not present in local branch
+    else()
+        set(patch_in_branch True PARENT_SCOPE)  # The patch is not present in local branch
+    endif()
+endfunction()
+
+# Validates if given SHA1/tag/branch name exists in local repo
+function(is_valid_revision repo_dir revision return_val)
+    message(STATUS "[OPENCL-CLANG] Validating ${revision} in repository")
+    # Check if we have under revision exitsting branch/tag/SHA1 in this repo
```
Comment (hewj03) [inline cmake/modules/CMakeFunctions.cmake:72]: typo exitsting

File: cmake/modules/CMakeFunctions.cmake [hunk 3]
```diff
@@ -39,6 +39,51 @@ macro(use_eh val)
...
+    execute_process(
+        COMMAND ${GIT_EXECUTABLE} log -1 ${revision}
+        WORKING_DIRECTORY ${repo_dir}
+        RESULT_VARIABLE output_var
+        ERROR_QUIET
+        OUTPUT_QUIET
+        )
+    if(${output_var} EQUAL 0)
+        set(${return_val} True PARENT_SCOPE) # this tag/branch/sha1 existis in repo
```
Comment (hewj03) [inline cmake/modules/CMakeFunctions.cmake:81]: typo existis

File: cmake/modules/CMakeFunctions.cmake [hunk 4]
```diff
@@ -39,6 +39,51 @@ macro(use_eh val)
...
+    else()
+        set(${return_val} False PARENT_SCOPE) # this tag/branch/sha1 not existis in repo
```
Comment (hewj03) [inline cmake/modules/CMakeFunctions.cmake:83]: typo existis

File: cmake/modules/CMakeFunctions.cmake (hunk 1 above)
Comment (lwesiers) [reply cmake/modules/CMakeFunctions.cmake:45]: I cannot see how I could read sha1 using git rev-parse from the file containing the patch before patching? Could you share the exact command line which you are thinking?

Comment (hewj03) [reply cmake/modules/CMakeFunctions.cmake:45]: I probably did a wrong test, rev-parse is not working.

**Review by hewj03 — APPROVED** [review]
LGTM

**Review by woshiyifei — APPROVED** [review]

**Pull request merged**